api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,161 @@
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Diagnosis class defines the identification of the nature of an illness or other problem 
+ * by examination of the symptoms during an encounter (visit or interaction of a patient with a healthcare worker).
+ * 
+ * @since 2.2
+ */
+public class Diagnosis extends BaseChangeableOpenmrsData {
+
+	private static final long serialVersionUID = 1L;
+	
+	private Integer diagnosisId;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+	
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+	
+	private Integer rank;
+	
+	public Diagnosis() {
+	}
+
+	/**
+	 * Diagnosis constructor with encounter, CodedOrFreeText, ConditionVerificationStatus, and rank
+	 *
+	 * @param encounter the encounter during which the diagnosis was made
+	 * @param diagnosis the CodedOrFreeText of the diagnosis
+	 * @param certainty the ConditionVerificationStatus of the diagnosis
+	 * @param rank the rank of the diagnosis
+	 */
+	public Diagnosis(Encounter encounter, CodedOrFreeText diagnosis, ConditionVerificationStatus certainty, Integer rank) {
+		this.encounter = encounter;
+		this.diagnosis = diagnosis;
+		this.certainty = certainty;
+		this.rank = rank;
+	}
+	
+	/**
+	 * Gets the diagnosisId.
+	 * 
+	 * @return the diagnosisId of the current diagnosis
+	 */
+	@Override
+	public Integer getId() {
+		return diagnosisId;
+	}
+
+	/**
+	 * Sets diagnosis diagnosisId
+	 * 
+	 * @param diagnosisId the diagnosisId to set 
+	 */
+	@Override
+	public void setId(Integer diagnosisId) {
+		this.diagnosisId = diagnosisId;
+	}
+
+	/**
+	 * Gets the encounter associated with this diagnosis.
+	 * 
+	 * @return encounter associated with this diagnosis
+	 */
+	public Encounter getEncounter() {
+		return encounter;
+	}
+
+	/**
+	 * Sets the encounter associated with this diagnosis.
+	 * 
+	 * @param encounter the encounter object to set
+	 */
+	public void setEncounter(Encounter encounter) {
+		this.encounter = encounter;
+	}
+
+	/**
+	 * Gets the diagnosis.
+	 * 
+	 * @return diagnosis the CodedOrFreeText associated with this diagnosis
+	 */
+	public CodedOrFreeText getDiagnosis() {
+		return diagnosis;
+	}
+
+	/**
+	 * Sets the CodedOrFreeText diagnosis.
+	 * 
+	 * @param diagnosis the CodedOrFreeText to set associated with this diagnosis
+	 */
+	public void setDiagnosis(CodedOrFreeText diagnosis)  {
+		this.diagnosis = diagnosis;
+	}
+
+	/**
+	 * Gets the diagnosis certainty.
+	 * 
+	 * @return certainty the ConditionVerificationStatus of this Diagnosis
+	 */
+	public ConditionVerificationStatus getCertainty() {
+		return certainty;
+	}
+
+	/**
+	 * Sets the diagnosis certainty
+	 * 
+	 * @param certainty the ConditionVerificationStatus to set for this Diagnosis
```

Comment:
The same applies to all the above where the calls names or types should now be included. For instance, just say: set condition verification status to set for this diagnosis (notice the small d for Diagnosis)